### ISSUE TITLE
Screen reader checkboxes

### ISSUE THREAD

**Issue by Norric1Admin** [ONSdigital/sdg-indicators#3079]
The checkboxes present within the sub-categories accordion menus do not indicate to screen reader users when they have been selected. Screen reader users would expect check boxes to announce ‘Checkbox checked’ and ‘Checkbox not checked’ respectively.

Update: We should fix the focus issue, so that the user isn't taken to chart.js when they check the box. The focus should remain on the checkbox (maybe speak to Brock, who was [looking at this code recently](https://github.com/open-sdg/open-sdg/issues/165)). Screenreader function should be tested as normal once this is solved.

**Comment by Norric1Admin** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Accessibility tickets](https://trello.com/c/E40VA79G/549-accessibility-tickets)

**Comment by SavvasStephanides** [comment]
I just did a quick test using VoiceOver on Safari. It does seem to notify the user whenever a checkbox is checked or unchecked ("ticked" / "unticked"). The checkbox does however seem to lose focus whenever it is checked / unchecked (I think the focus goes to "Clear selections")

**Comment by Norric1Admin** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Accessibility: Sprint 12](https://trello.com/c/HZJO8O4J/576-accessibility-sprint-12)

**Comment by Norric1Admin** [comment]
@SavvasStephanides Is this related to https://github.com/open-sdg/open-sdg/pull/162#pullrequestreview-234978979?

**Comment by Norric1Admin** [comment]
From DAC:
For ‘Checkboxes State’, you recommend adding an audio label so screenreaders can recognise when a category is selected. Our developer tried this on Safari and found that this already exists. Please let me know which browser and screenreader was used
"JAWS v.16/18 with Internet Explorer 11 and NVDA with Mozilla Firefox."

When a checkbox is selected, the focus shifts from categories to the table. Where should the ideal focus be after selecting?
"The focus should remain on the checkbox but users must be informed that it has been checked "

**Comment by LucyGwilliamAdmin** [comment]
## Description

## Screenshots
**Before**

**After**


## Successfully tested on...
- [ ] Windows: IE11
- [ ] Windows: Chrome
- [ ] Windows: Firefox
- [ ] Windows: Edge
- [ ] macOS: Chrome
- [ ] macOS: Firefox
- [ ] iOS: Safari 9.3
- [ ] iOS: Chrome
- [ ] Samsung: Chrome
- [ ] Samsung: Samsung Internet

## User feedback and problems encountered...

**Comment by SavvasStephanides** [comment]
## Update

As mentioned before, the reason screen readers don't announce when a checkbox has been ticked or unticked, is because the checkbox loses focus after the change, thus the screen reader moving on to announcing the next focused item and "forgetting" about the checkbox status. The default behaviour in these cases is that the checkbox keeps the focus when ticked or unticked ([see example here](https://codepen.io/anon/pen/zQJMaq?editors=1000)). Something must have therefore intervened with the default behaviour.

Most of last week and today I have been investigating which piece of code could have caused this unexpected shift in focus. It was like finding a needle in a haystack mostly because of the complexity of the code. So far I have concluded that the problem is somewhere along [these lines](https://github.com/open-sdg/open-sdg/blob/master/_includes/assets/js/indicatorModel.js#L339-L578). Still a long way to go.

I'll update further soon.

**Comment by SavvasStephanides** [comment]
Pull request: https://github.com/ONSdigital/sdg-indicators/pull/3156

**Comment by Norric1Admin** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Screen reader checkboxes](https://trello.com/c/t1PWW5Wo/603-screen-reader-checkboxes)

**Comment by Norric1Admin** [comment]
@SavvasStephanides The only problem with this is (I think) the ARIA tab reading 'space' as "space", rather than the expected "Checkbox checked/unchecked".
Try this?: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role

```
function changeCheckbox(event) {
    let item = document.getElementById('chkPref');
    switch(item.getAttribute('aria-checked')) {
        case "true":
            item.setAttribute('aria-checked', "false");
            break;
        case "false":
            item.setAttribute('aria-checked', "true");
            break;
    }
}
```

**Comment by Norric1Admin** [comment]
We can decide whether to pursue getting this working on IE / Windows Narrator (the only faulty combo).
